lib/najax.js

```diff
@@ -168,28 +159,35 @@ function najax (uri, options, callback) {
       jqXHR.status = statusCode
 
       if (isSuccess) {
-        // Set data for the fake xhr object
-        jqXHR.statusText = statusText
         // success, statusText, jqXHR
-        dfd.resolve(data, statusText, jqXHR)
+        dfd.resolve(data, jqXHR.statusText, jqXHR)
       } else {
         // jqXHR, statusText, error
         // When an HTTP error occurs, errorThrown receives the textual portion of the
         // HTTP status, such as "Not Found" or "Internal Server Error."
-        statusText = 'error'
-        dfd.reject(jqXHR, statusText, http.STATUS_CODES[statusCode])
+        jqXHR.statusText = 'error'
+        onError(new Error(http.STATUS_CODES[statusCode]))
       }
     })
   })
 
   // ERROR
-  req.on('error', errorHandler)
+  req.on('error', onError)
+
+  function onError (e) {
+    // Set data for the fake xhr object
+    if (jqXHR.statusText === 'error') jqXHR.responseText = e.stack
+    if (_.isFunction(o.error)) o.error(jqXHR, jqXHR.statusText, e)
```

Comment:
Since I added `o.error` to the `dfd.reject` stack at [lib/najax.js#L203](https://github.com/najaxjs/najax/blob/master/lib/najax.js#L203), we won't need this line 180 any longer.